src/components/VideoProvider/useLocalTracks/useLocalTracks.tsx

```diff
@@ -1,137 +1,96 @@
-        console.log('No microphone attached.');
-      });
-  }, [defaultDevice]);
-
-  useEffect(() => {
-    Video.createLocalAudioTrack({
-      deviceId: selectedAudioInput.deviceId,
-      groupId: selectedAudioInput.groupId,
-    })
-      .then((newTrack: any) => {
-        setTrack(newTrack);
-      })
-      .catch(err => {
-        console.log('No microphone attached.');
-      });
-  }, [selectedAudioInput]);
-
-  useEffect(() => {
-    const handleStopped = () => setTrack(undefined);
-    if (track) {
-      track.on(PLAYER_STATE.stopped, handleStopped);
-      return () => {
-        track.off(PLAYER_STATE.stopped, handleStopped);
-      };
-    }
-  }, [track]);
-
-  return track;
-}
-
-export function useLocalVideoTrack() {
-  const [track, setTrack] = useState<any>();
-
-  const { selectedVideoInput } = useAppState();
-
-  const getLocalVideoTrack = useCallback(
-    () =>
-      Video.createLocalVideoTrack({
-        deviceId: selectedVideoInput.deviceId,
-        groupId: selectedVideoInput.groupId,
-        frameRate: 24,
-        height: 720,
-        width: 1280,
-      })
-        .then((newTrack: any) => {
-          setTrack(newTrack);
-          return newTrack;
-        })
-        .catch(err => {
-          if (err.message === 'Requested device not found') {
-            console.log('No camera attached.');
-          }
-        }),
-    [selectedVideoInput]
-  );
-
-  useEffect(() => {
-    // We get a new local video track when the app loads.
-    getLocalVideoTrack();
-  }, [selectedVideoInput, getLocalVideoTrack]);
-
-  useEffect(() => {
-    console.log('manual', selectedVideoInput.label);
-  }, [selectedVideoInput]);
-
-  useEffect(() => {
-    const handleStopped = () => setTrack(undefined);
-    if (track) {
-      track.on(PLAYER_STATE.stopped, handleStopped);
-      return () => {
-        track.off(PLAYER_STATE.stopped, handleStopped);
-      };
-    }
-  }, [track]);
-
-  return [track, getLocalVideoTrack];
-}
-
-export default function useLocalTracks() {
-  const getLocalAudioTrack = useLocalAudioTrack();
-  const [videoTrack, getLocalVideoTrack] = useLocalVideoTrack();
-  const [isAcquiringLocalTracks, setIsAcquiringLocalTracks] = useState(false);
-  const [Stat, setVideoTrack] = useState();
-  const removeLocalVideoTrack = useCallback(() => {
-    if (videoTrack) {
-      videoTrack.stop();
-      setVideoTrack(undefined);
-    }
-  }, [videoTrack]);
-
-  const localTracks: (LocalAudioTrack | LocalVideoTrack)[] = [getLocalAudioTrack, videoTrack].filter(
-    track => track !== undefined
-  );
-  return { localTracks, getLocalVideoTrack, getLocalAudioTrack, isAcquiringLocalTracks, removeLocalVideoTrack };
-}
+import { DEFAULT_VIDEO_CONSTRAINTS } from '../../../constants';
+import { useCallback, useEffect, useState } from 'react';
+import Video, { LocalVideoTrack, LocalAudioTrack, CreateLocalTrackOptions } from 'twilio-video';
+import { TRACK_TYPE } from '../../../utils/displayStrings';
+

```

Comment:
Why we removed functionality?

Reply:
I did not remove, it's just unused imports.